src/main/java/organice/logic/commands/EditCommand.java

```diff
@@ -90,6 +117,28 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         Nric updatedNric = editPersonDescriptor.getNric().orElse(personToEdit.getNric());
         Name updatedName = editPersonDescriptor.getName().orElse(personToEdit.getName());
         Phone updatedPhone = editPersonDescriptor.getPhone().orElse(personToEdit.getPhone());
+        if (updatedType.isPatient()) {
+            Age updatedAge = ((Patient) personToEdit).getAge();
+            BloodType updatedBloodType = ((Patient) personToEdit).getBloodType();
+            TissueType updatedTissueType = ((Patient) personToEdit).getTissueType();
+            Organ updatedOrgan = ((Patient) personToEdit).getOrgan();
+            Priority updatedPriority = ((Patient) personToEdit).getPriority();
+            DoctorInCharge updatedDoctorInCharge = ((Patient) personToEdit).getDoctorInCharge();
+            Status updatedStatus = ((Patient) personToEdit).getStatus();
+            return new Patient(updatedType, updatedNric, updatedName, updatedPhone, updatedAge, updatedPriority,
+                    updatedBloodType, updatedTissueType, updatedOrgan, updatedDoctorInCharge, updatedStatus);
+        } else if (updatedType.isDonor()) {
+            Age updatedAge = ((Donor) personToEdit).getAge();
+            BloodType updatedBloodType = ((Donor) personToEdit).getBloodType();
+            TissueType updatedTissueType = ((Donor) personToEdit).getTissueType();
+            Organ updatedOrgan = ((Donor) personToEdit).getOrgan();
+            OrganExpiryDate updatedOrganExpiryDate = ((Donor) personToEdit).getOrganExpiryDate();
+            Status updatedStatus = ((Donor) personToEdit).getStatus();
+            return new Donor(updatedType, updatedNric, updatedName, updatedPhone, updatedAge,
+                updatedBloodType, updatedTissueType, updatedOrgan, updatedOrganExpiryDate, updatedStatus);
+        } else if (updatedType.isDoctor()) {
+            return new Doctor(updatedType, updatedNric, updatedName, updatedPhone);
+        }
 
         return new Person(updatedType, updatedNric, updatedName, updatedPhone);
```

Comment:
Perhaps you can throw an error instead of returning a `Person`? Because if the type is not patient, donor or doctor, then it is an invalid type. 

Another way to do this could be to add an assertion. To assert that the type is Patient, Donor or Doctor. Because there is type checking when creating the `Type` object. So the assumption is that the type will be a valid type.